### PR TITLE
feat: added the properties attribute to the pageview

### PR DESCRIPTION
**Pull request by aswathy-deriv**

### CODE REVIEW

File: src/rudderstack.ts [hunk 1]
```diff
@@ -125,9 +125,9 @@ export class RudderStack {
      *
      * @param curret_page The name or URL of the current page to track the page view event
      */
-    pageView = (current_page: string, platform = 'Deriv App', user_id: string) => {
+    pageView = (current_page: string, platform = 'Deriv App', user_id: string, properties?: {}) => {
         if (this.has_initialized && this.has_identified && current_page !== this.current_page) {
-            this.analytics.page(platform, current_page, { user_id })
+            this.analytics.page(platform, current_page, { user_id }, properties)
```
Comment (prince-deriv) [inline src/rudderstack.ts:130]: you need to merge the properties in the 3rd parameter

```suggestion
            this.analytics.page(platform, current_page, { user_id, ...properties })
```

Comment (prince-deriv) [reply src/rudderstack.ts:130]: 4th parameter is for `options` if I'm not mistaken.

Comment (aswathy-deriv) [reply src/rudderstack.ts:130]: @prince-deriv Actually,the pageView coming from rudderstack is having `properties` as 3rd attribute and 4th one is `options` which we are not using.But we are pushing one more attribute as `userId` into our analytics and then the properties as 4th we can pass it as whatever needed.So ideally while calling the pageview
```
eg:
   Analytics.pageView(window.location.href, {
            loggedIn: true,
            device_type: 'desktop',
        });
```
we are not passing the userId while calling the pageview.
The properties will have custom values which we are passing
Screenshot 2024-10-25 at 11 02 26 AM

Comment (ali-hosseini-deriv) [reply src/rudderstack.ts:130]: Actually I also think @prince-deriv is right.
both of them should be passed in one object.
Check their docs https://www.rudderstack.com/docs/event-spec/standard-events/page/#page-fields for more info.

Comment (ali-hosseini-deriv) [reply src/rudderstack.ts:130]: Now i get it, the userID was always outside and on the root level, so now if we change this schema then cro squad also should change their queries too.
So we better to stick with current implementation.

**Review by ali-hosseini-deriv — APPROVED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**